remote.go

```diff
@@ -728,6 +728,39 @@ func (r *Remote) buildFetchedTags(refs memory.ReferenceStorage) (updated bool, e
 	return
 }
 
+// LSRemote performs ls-remote on the remote.
+func (r *Remote) LSRemote(auth transport.AuthMethod) ([]*plumbing.Reference, error) {
```

Comment:
Maybe we should call this method `List`? 
```
// List the references on the remote repository
```
Thoughts @mcuadros?

Reply:
Also, does it make sense to generalize the parameters so that it accepts an Options-style struct instead of just `transport.AuthMethod` to help ease future changes without breaking API compatibility?

Reply:
Yep, a Options will be the best

Reply:
Yep, makes sense. I created an Option at first, but then dropped it thinking it would be simpler to just pass the `AuthMethod` directly 😅. Adding it back...